Source/WebCore/platform/adwaita/ScrollbarThemeAdwaita.cpp

```
@@ -323,14 +328,8 @@ ScrollbarButtonPressAction ScrollbarThemeAdwaita::handleMousePressEvent(Scrollba
     switch (pressedPart) {
     case BackTrackPart:
     case ForwardTrackPart:
-#if PLATFORM(GTK)
-        warpSlider = [] {
-            gboolean gtkWarpsSlader = FALSE;
-            g_object_get(gtk_settings_get_default(),
-                "gtk-primary-button-warps-slider",
-                &gtkWarpsSlader, nullptr);
-            return gtkWarpsSlader;
-        }();
+#if PLATFORM(GTK) || PLATFORM(WPE)
+        warpSlider = SystemSettingsGLib::singleton().primaryButtonWarpsSlider().value_or(true);
```

Why true by default? It was false before, no?

---

It was never false by default, it was just misleading code.

`g_object_get()` always overwrites the value passed in, and in GTK the default value was TRUE.